In [1]:
import pandas as pd

df = pd.read_csv("../data/raw/creditcard.csv")

print("Dataset Shape:", df.shape)
df.head()

PermissionError: [Errno 13] Permission denied: '../data/raw/creditcard.csv'

In [2]:
import os

file_path = "../data/raw/creditcard.csv"

print("File exists:", os.path.exists(file_path))
print("File path:", os.path.abspath(file_path))
print("File size:", os.path.getsize(file_path))

File exists: True
File path: c:\Users\Amaan\OneDrive\Desktop\real-time-financial-fraud-detection\data\raw\creditcard.csv
File size: 0


In [3]:
DATA_PATH = "../data/raw/creditcard.csv/creditcard.csv"

df = pd.read_csv(DATA_PATH)

print("Dataset loaded successfully!")
print("Total transactions:", len(df))

Dataset loaded successfully!
Total transactions: 284807


In [4]:
print("Dataset Shape:", df.shape)

print("\nMissing Values:")
print(df.isnull().sum().sum())

print("\nDuplicate Rows:")
print(df.duplicated().sum())


Dataset Shape: (284807, 31)

Missing Values:
0

Duplicate Rows:
1081


In [5]:
df = df.drop_duplicates()

print("Duplicates removed successfully!")
print("New Dataset Shape:", df.shape)
print("Remaining Duplicates:", df.duplicated().sum())

Duplicates removed successfully!
New Dataset Shape: (283726, 31)
Remaining Duplicates: 0


In [6]:
OUTPUT_PATH = "../data/processed/creditcard_clean.csv"

df.to_csv(OUTPUT_PATH, index=False)

print("Cleaned dataset saved successfully!")
print("Saved rows:", len(df))
print("Saved path:", OUTPUT_PATH)

Cleaned dataset saved successfully!
Saved rows: 283726
Saved path: ../data/processed/creditcard_clean.csv


In [7]:
print("Final Dataset Shape:", df.shape)
print("Missing Values:", df.isnull().sum().sum())
print("Duplicate Rows:", df.duplicated().sum())

print("\nClass Distribution:")
print(df["Class"].value_counts())


Final Dataset Shape: (283726, 31)
Missing Values: 0
Duplicate Rows: 0

Class Distribution:
Class
0    283253
1       473
Name: count, dtype: int64


## Data Cleaning Findings

- The original dataset contained 284,807 transactions and 31 columns.
- No missing values were found in the dataset.
- 1,081 duplicate transactions were identified and removed.
- After cleaning, the dataset contains 283,726 unique transactions.
- No duplicate rows or missing values remain.
- The cleaned dataset contains 283,253 non-fraud transactions and 473 fraud transactions.
- The cleaned dataset was saved as `data/processed/creditcard_clean.csv`.
